packages/material-ui/src/ButtonBase/ButtonBase.d.ts

```diff
@@ -72,11 +72,10 @@ export interface ButtonBaseTypeMap<P = {}, D extends React.ElementType = 'button
      * The system prop that allows defining system overrides as well as additional CSS styles.
      */
     sx?: SxProps<Theme>;
-    // @types/react is stricter
     /**
      * @default 0
      */
-    tabIndex?: string | number;
+    tabIndex?: NonNullable<React.HTMLAttributes<any>['tabIndex']>;
```

Comment:
We are using `tabIndex?: number;` in the select and slider component. Which option is better? Should we update t these two components too?